addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4126,7 +4195,7 @@ export class OdooEditor extends EventTarget {
             if (ev.shiftKey) {
                 this._resetLinkZws();
             }
-            getDeepRange(this.editable, { select: true });
+            handleMultipleRanges(this.editable, { select: true, correctTripleClick: selection.anchorNode && selection.anchorNode.nodeName === 'TR' });
```

Comment:
The code below does not handle table selection then why did you had to handle multile ranges here?
Same for https://github.com/odoo/odoo/pull/147880/commits/f65c87279348e00f61a3bf2c3b1f5a37167d995c#diff-c309d19284e2e1f38a1dba1d68e0262371686595f3668a3bc0f1ca35e3189a0bR4242 ?

Reply:
We need to do that to correct the selection before `_setSelectionInTable` gets called. In firefox, when multiple cells are selected through mouse and then if we press `shift + ArrowRight` then it results in wrong selection. I happens because `getDeepRange(this.editable, { select: true })` gets called before `_setSelectionInTable`.

Reply:
We can give a try :+1: